sfe/sfe.go

```diff
@@ -0,0 +1,317 @@
+	// to JIT parse. The pages are stored in an in-memory embed.FS to prevent
+	// unnecessary filesystem I/O on a physical HDD.
+	tmplPages := template.Must(template.New("pages").ParseFS(dynamicFS, "templates/layout.html", "pages/*"))
+
+	sfe := SelfServiceFrontEndImpl{
+		log:             logger,
+		clk:             clk,
+		requestTimeout:  requestTimeout,
+		ra:              rac,
+		sa:              sac,
+		unpauseHMACHash: unpauseHMACHash,
+		templatePages:   tmplPages,
+	}
+
+	return sfe, nil
+}
+
+// Handler returns an http.Handler that uses various functions for various
+// non-ACME-specified paths. Each endpoint should have a corresponding HTML
+// page that shares the same name as the endpoint.
+func (sfe *SelfServiceFrontEndImpl) Handler(stats prometheus.Registerer, oTelHTTPOptions ...otelhttp.Option) http.Handler {
+	m := http.NewServeMux()
+
+	sfs, _ := fs.Sub(staticFS, "static")
+	staticAssetsHandler := http.StripPrefix("/static/", http.FileServerFS(sfs))
+
+	m.Handle("GET /static/", staticAssetsHandler)
+	m.HandleFunc("/", sfe.Index)
+	m.HandleFunc("GET /build", sfe.BuildID)
+	m.HandleFunc(unpauseGetForm, sfe.Unpause)
+	m.HandleFunc(unpausePostForm, sfe.Unpause)
+	m.HandleFunc(unpauseSuccessful, sfe.UnpauseSuccessful)
+
+	return measured_http.New(m, sfe.clk, stats, oTelHTTPOptions...)
+}
+
+// renderTemplate takes an HTML template instantiated by the SFE init() and an
+// optional dynamicData which are rendered and served back to the client via the
+// response writer.
+func (sfe *SelfServiceFrontEndImpl) renderTemplate(w http.ResponseWriter, filename string, dynamicData any) {
+	if len(filename) == 0 {
+		http.Error(w, "Template page does not exist", http.StatusInternalServerError)
+		return
+	}
+
+	err := sfe.templatePages.ExecuteTemplate(w, filename, dynamicData)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+}
+
+// Index is the homepage of the SFE
+func (sfe *SelfServiceFrontEndImpl) Index(response http.ResponseWriter, request *http.Request) {
+	if request.Method != http.MethodGet && request.Method != http.MethodHead {
+		response.WriteHeader(http.StatusMethodNotAllowed)
+		return
+	}
+	sfe.renderTemplate(response, "index.html", nil)
+}
+
+// BuildID tells the requester what boulder build version is running.
+func (sfe *SelfServiceFrontEndImpl) BuildID(response http.ResponseWriter, request *http.Request) {
+	response.Header().Set("Content-Type", "text/plain")
+	response.WriteHeader(http.StatusOK)
+	detailsString := fmt.Sprintf("Boulder=(%s %s)", core.GetBuildID(), core.GetBuildTime())
+	if _, err := fmt.Fprintln(response, detailsString); err != nil {
+		sfe.log.Warningf("Could not write response: %s", err)
+	}
+}
+
+// unpauseJWT is generated by a WFE and is used to round-trip back through the
+// SFE to unpause the requester's account.
+type unpauseJWT string
+
+func (sfe *SelfServiceFrontEndImpl) getHelper(response http.ResponseWriter, incomingJWT unpauseJWT) {
+	if incomingJWT != "" {
```

Comment:
Invert the logic here and in postHelper:
```
if incomingJWT == "" {
  render error page
}

render real page
```

This way you don't have a big block of code above an `else`; just get the error case out of the way and then do straight-line happy-path code below that.